docs: Changelog and Third Party for 0.1.0 (#233)
Unity-Technologies/com.unity.multiplayer.samples.coop: Packages/com.unity.multiplayer.samples.coop/CHANGELOG.md

# Multiplayer Samples Co-op Changelog

## [0.1.0] - 2021-04-07

v0.1.0 is an Early Access release for Multiplayer Samples Co-op.

It requires and supports Unity v2020LTS and Unity MLAPI v0.1.0. For additional information on MLAPI, see the changelog and release notes for the Unity MLAPI package.

### New features

The Multiplayer Samples Co-op release provides Boss Room, a small-scale cooperative game sample project built on top of the new experimental netcode library. The release provides example code, assets, and integrations to explore the concepts and patterns behind a multiplayer game flow. It supports up to 8 players for testing multiplayer functionality.

> * See the README for installation instructions, available in the downloaded release and [GitHub](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.coop).

Reviewer: ```suggestion
* See the README for installation instructions, available in the downloaded [release](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.coop/releases/latest) and [GitHub](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.coop).
```

# Multiplayer Samples Co-op Changelog

## [0.1.0-experimental] - 2021-04-07

v0.1.0 is the first, experimental release for Multiplayer Samples Co-op.

Requires and supports Unity version 2019.4 and later.

### New features

The Multiplayer Samples Co-op release provides Boss Room, a small-scale cooperative game sample project built on top of the new experimental netcode library. The release provides example code, assets, and integrations to explore the concepts and patterns behind a multiplayer game flow. It supports up to 8 players for testing multiplayer functionality.

* Access Boss Room by downloading the project and adding it using Package Manager.
* Learn with Unity using the Boss Room source code, project files, and assets which include: 1 populated dungeon level, 4 character classes with 2 genders, combatant imps and boss, and a simple collaborative puzzle.
* Test co-op games by running multiple instances of the game locally or connecting to a friend over the internet.

### Known issues

> * After a round of Boss Room completes and a party member returns to the menu, it pulls all players and leader to menu.

Reviewer: as discussed today, GOMPS-431 should still be confirmed, this was fixed previously

# Multiplayer Samples Co-op Changelog

## [0.1.0] - 2021-04-07

v0.1.0 is an Early Access release for Multiplayer Samples Co-op.

It requires and supports Unity v2020.3 and later and Unity MLAPI v0.1.0. For additional information on MLAPI, see the changelog and release notes for the Unity MLAPI package.

### New features

> The Multiplayer Samples Co-op release provides Boss Room, a small-scale cooperative game sample project built on top of the new experimental netcode library. The release provides example code, assets, and integrations to explore the concepts and patterns behind a multiplayer game flow. It supports up to 8 players for testing multiplayer functionality.

Reviewer 2: ```suggestion
Boss Room is a small-scale cooperative game sample project built on top of the new experimental netcode library. The release provides example code, assets, and integrations to explore the concepts and patterns behind a multiplayer game flow. It supports up to 8 players for testing multiplayer functionality.
```

### Known issues

The following issues may occur for access and games:

* Sometimes when the host leaves the Boss Room, not all clients will return to the Main Menu, but will remain stuck in the Boss Room scene. 
* Sometimes after completing a match and the host starts a new match from the Victory or Loss screen, connected players may have no visible interactions to join or select characters.
* A player may encounter a rare exception when the Tank character uses her Shield Aura ability. This issue may be due to intercepting the Boss charge attack.
* If two players in the Character Select **Ready** for the same hero at the same time, the UI will update to *Readied* on both clients, but only one will have actually selected the hero on the Host. This issue blocks Character Select from proceeding.
* Any player that plays the game and then returns to the Main Menu may be unable to Start or Join properly again, requiring you to restart the client.

> * The Green Quads may show on impact when the Archer arrow strikes enemies. This issue may only occur in the editor.

Reviewer: ```suggestion
* Green quads may show on impact when the Archer arrow strikes enemies. This issue may only occur in the editor.
```

* You may encounter a game error due to a Unity MLAPI exception. Not all MLAPI exceptions are fully exposed few informing users.
* The Photon Transport currently generates some errors in the Player log related to the `PhotonCryptoPlugin`.

> * The welcome player message in the lobby indicates P2 (player 2) regardless of your player number. Currently the Character Select scene displays “Player1” and “P1” in two locations, where it is intended that the user’s name be displayed.  

Reviewer: ```suggestion
* The welcome player message in the lobby indicates P2 (player 2) regardless of your generated name. Currently the Character Select scene displays “Player1” and “P1” in two locations, where it is intended that the user’s name be displayed.  
```

> * The spawner portal system requires updates to better handle wave spawning logic and destroying crystal game elements.

Reviewer: ```suggestion
* The spawner portal don't work in this release.
```

> * Players may not reliably play another match when selecting **Return to Main Menu** during the post-game scene. THis may be due to states not properly clearing.

Reviewer: ```suggestion
* Players may not reliably play another match when selecting **Return to Main Menu** during the post-game scene. This may be due to states not properly clearing.
```

> 

Reviewer: ```suggestion
Some actions may feel unresponsive and require action anticipation animations
```
